src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -97,144 +116,113 @@ private static boolean matchesAuthor(PGMMap map, String query) {
       }
     }
 
     audience.sendMessage(
         new PersonalizedText(
             mapInfoLabel("command.map.mapInfo.playerLimit"),
-            createPlayerLimitComponent(sender, persistentContext)));
+            createPlayerLimitComponent(sender, map)));
 
     if (sender.hasPermission(Permissions.DEBUG)) {
-      audience.sendMessage(
-          new PersonalizedText(
-              mapInfoLabel("command.map.mapInfo.genre"),
-              new PersonalizedText(mapInfo.getLocalizedGenre(), ChatColor.GOLD)));
       audience.sendMessage(
           new PersonalizedText(
               mapInfoLabel("command.map.mapInfo.proto"),
-              new PersonalizedText(mapInfo.proto.toString(), ChatColor.GOLD)));
-      audience.sendMessage(
-          new PersonalizedText(
-              mapInfoLabel("command.map.mapInfo.folder"),
-              new PersonalizedText(map.getFolder().getRelativePath().toString(), ChatColor.GOLD)));
-      audience.sendMessage(
-          new PersonalizedText(
-              mapInfoLabel("command.map.mapInfo.source"),
-              new PersonalizedText(map.getSource().getPath().toString(), ChatColor.GOLD)));
+              new PersonalizedText(map.getProto().toString(), ChatColor.GOLD)));
     }
 
-    URL xmlLink = map.getFolder().getDescriptionFileUrl();
-    if (xmlLink != null) {
-      audience.sendMessage(
-          new PersonalizedText(
-              new PersonalizedText(ChatColor.DARK_PURPLE, ChatColor.BOLD)
-                  .extra(new PersonalizedTranslatable("command.map.mapInfo.xml"))
-                  .extra(": "),
-              Components.link(xmlLink)
-                  .hoverEvent(
-                      HoverEvent.Action.SHOW_TEXT,
-                      new PersonalizedTranslatable("command.map.mapInfo.sourceCode.tip")
-                          .render())));
-    }
-
-    Set<MapTag> mapTags = persistentContext.getMapTags();
-    audience.sendMessage(createTagsComponent(mapTags).color(ChatColor.YELLOW));
+    audience.sendMessage(createTagsComponent(map.getTags()));
   }
 
-  private static Component createTagsComponent(Set<MapTag> tags) {
+  private Component createTagsComponent(Collection<MapTag> tags) {
     checkNotNull(tags);
 
-    Component result = new PersonalizedText();
+    Component result = mapInfoLabel("command.map.mapInfo.tags");
     MapTag[] mapTags = tags.toArray(new MapTag[0]);
     for (int i = 0; i < mapTags.length; i++) {
       if (i != 0) {
         result.extra(Components.space());
       }
 
-      MapTag mapTag = mapTags[i];
+      String mapTag = mapTags[i].getId();
       Component component =
-          mapTags[i]
-              .getComponentName()
-              .clickEvent(ClickEvent.Action.RUN_COMMAND, "/maplist " + mapTag.toString())
+          new PersonalizedText("#" + mapTag, ChatColor.GOLD)
+              .bold(false)
+              .clickEvent(ClickEvent.Action.RUN_COMMAND, "/maps -t " + mapTag)
```

Comment:
`/maps` is an alias to `/maplist` so I would keep `/maplist` here. :P

Reply:
`/maps` is the main command.